packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -446,24 +446,58 @@ function disposeViewRemoteVideo(
 }
 
 function disposeViewLocalVideo(context: CallContext, internalContext: InternalCallContext, callId: string): void {
+  const streamLogInfo = { callId };
+
+  _logEvent(callingStatefulLogger, {
+    name: EventNames.START_DISPOSE_LOCAL_STREAM,
+    level: 'info',
+    message: 'Start disposing local stream.',
+    data: streamLogInfo
+  });
+
   context.setLocalVideoStreamRendererView(callId, undefined);
 
   const renderInfo = internalContext.getLocalRenderInfo(callId);
   if (!renderInfo) {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_DISPOSE_INFO_NOT_FOUND,
+      level: 'error',
+      message: 'Cannot find render info when disposing local stream.',
+      data: streamLogInfo
+    });
     return;
   }
 
   // Sets the status and also renderer. I think we need to always set renderer to undefined since in all status when
   // cleaned up should have renderer as undefined. If the status is 'Rendered' and renderer is not defined it should
   // be cleaned up below so we can set it to undefined here.
   if (renderInfo.status === 'Rendering') {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_STREAM_STOPPING,
+      level: 'warning',
+      message: 'Local stream is still rendering. Changing status to stopping.',
+      data: streamLogInfo
+    });
     internalContext.setLocalRenderInfo(callId, renderInfo.stream, 'Stopping', undefined);
   } else {
     internalContext.setLocalRenderInfo(callId, renderInfo.stream, 'NotRendered', undefined);
   }
 
   if (renderInfo.renderer) {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.DISPOSING_LOCAL_RENDERER,
+      level: 'info',
+      message: 'Disposing local view renderer.',
+      data: streamLogInfo
+    });
     renderInfo.renderer.dispose();
+  } else {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_RENDERER_NOT_FOUND,
+      level: 'warning',
```

Comment:
Shouldn't this also be an `error`, similar to line 464?

Reply:
Yep I agree, updating here and remote rendering ops to be error